test/main_contracts/rate_oracles/compoundBorrowRateOracle.ts

```diff
@@ -0,0 +1,63 @@
+import { ethers, waffle } from "hardhat";
+import { BigNumber, Wallet } from "ethers";
+import { expect } from "chai";
+import { ConfigForGenericTests as Config } from "./compoundBorrowConfig";
+import {
+  MockCToken,
+  TestCompoundBorrowRateOracle,
+  MockCInterestRateModel,
+} from "../../../typechain";
+
+const { provider } = waffle;
+
+describe("Compound Borrow Rate Oracle", () => {
+  const blocksPerYear = BigNumber.from(31536000).div(13);
+  const wad = BigNumber.from(10).pow(18);
+  const ratePerBlock = BigNumber.from(2).mul(wad).div(100).div(blocksPerYear);
+
+  let wallet: Wallet, other: Wallet;
+  let loadFixture: ReturnType<typeof waffle.createFixtureLoader>;
+  let testCompoundBorrowRateOracle: TestCompoundBorrowRateOracle;
+  let cToken: MockCToken;
+
+  before("create fixture loader", async () => {
+    [wallet, other] = await (ethers as any).getSigners();
+    loadFixture = waffle.createFixtureLoader([wallet, other]);
+  });
+
+  describe("Compound Borrow specific behaviour", () => {
+    beforeEach("deploy and initialize test oracle", async () => {
+      const { testRateOracle } = await loadFixture(Config.oracleFixture);
+      testCompoundBorrowRateOracle =
+        testRateOracle as unknown as TestCompoundBorrowRateOracle;
+      cToken = (await ethers.getContract("MockCToken")) as MockCToken;
+
+      const mockCInterestRateModel = (await ethers.getContract(
+        "MockCInterestRateModel"
+      )) as MockCInterestRateModel;
+
+      await mockCInterestRateModel.setBorrowRatePerBlock(ratePerBlock);
+    });
+    it("Verify correct protocol ID for Compound Borrow rate oracle", async () => {
+      const protocolID =
+        await testCompoundBorrowRateOracle.UNDERLYING_YIELD_BEARING_PROTOCOL_ID();
+
+      expect(protocolID).to.eq(2);
+    });
+
+    it(`Verify rate for Compound Borrow`, async () => {
```

Comment:
The logic for `getLastUpdatedRate` is more involved than other implementations and if it stays that way there is probably more we should test here. E.g. testing with a variety of different blockDelta values, including 0 and 1 as well as large numbers.

Reply:
agreed, @CyclopsRex can you pls write up a set of edge case scenarios you'd personally test in this instance? the main difference between compound and aave borrowing rate oracles is we manually have to do borrow index extrapolation using the current borrow rate + block delta between the latest accrual on compound and current block

Reply:
The thing that's new/different is this extrapolation by blocks-since-last-accrual, so I would test that for a variety of `blockDelta` values, including `blockDelta = 0` and `blockDelta = 1`. Check that the math works out as we expect.

Reply:
test cases included